persistence/sql/persister_jwk.go

```diff
@@ -62,6 +81,30 @@ func (p *Persister) AddKeySet(ctx context.Context, set string, keys *jose.JSONWe
 	})
 }
 
+func (p *Persister) UpdateKey(ctx context.Context, set string, key *jose.JSONWebKey) error {
+	if err := p.DeleteKey(ctx, set, key.KeyID); err != nil {
+		return err
+	}
+
+	if err := p.AddKey(ctx, set, key); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (p *Persister) UpdateKeySet(ctx context.Context, set string, keySet *jose.JSONWebKeySet) error {
+	if err := p.DeleteKeySet(ctx, set); err != nil {
```

Comment:
Similar to UpdateKeySet this needs to be in a transaction and it needs a test :)